### PR TITLE
Resolved issue with logfile being created even if no logging was done

### PR DESCRIPTION
**Pull request by modelonrobinandersson**

### CODE REVIEW

File: src/pyfmi/fmi.pyx [hunk 1]
```diff
@@ -625,14 +631,15 @@ cdef class ModelBase:
...
+        """ Sets parameter to keep log file open when logging is done to log file, if we are not logging into a given stream. """
         if not self._log_is_stream and self._fmu_log_name != NULL:
-            self.file_object = open(self._fmu_log_name,'a')
+            self._keep_log_open = 1
 
     def _close_log_file(self):
         if self.file_object:
             self.file_object.close()
             self.file_object = None
+            self._keep_log_open = 0
```
Comment (chria) [inline src/pyfmi/fmi.pyx:642]: I propse to always set this in this function. I.e. move self._keep_log_open = 0 outside of the if-statement

File: tests/test_log.py [hunk 2]
```diff
@@ -100,3 +105,62 @@ def test_extract_boolean_value(self):
         eis = log.find("EventInfo")
         for ei in eis:
             assert isinstance(ei.time_event_info, bool), "Expected ei.time_event_info to be bool"
+
+class TestNoLogFile:
```
Comment (chria) [inline tests/test_log.py:109]: Does these tests really work?

**Review by chria — COMMENTED** [review]
Looks good! Just a few comments

File: src/pyfmi/fmi.pyx (hunk 1 above)
Comment (modelonrobinandersson) [reply src/pyfmi/fmi.pyx]: Good point, I've updated the code with the latest commit

File: tests/test_log.py (hunk 2 above)
Comment (modelonrobinandersson) [reply tests/test_log.py:109]: Yes they do, they run without issues when I run them locally, or is it anything you see that indicates that they should not work?

File: tests/test_log.py [hunk 3]
```diff
@@ -100,3 +105,62 @@ def test_extract_boolean_value(self):
...
+    """
+        Test invoking functions on the FMU if we have no logfile.
+        We do this from temporary directories to make sure that we have empty directories for the test.
+    """
+    @classmethod
+    def setup_class(cls):
+        cls.fmu_path = os.path.join(get_examples_folder(), 'files', 'FMUs', 'ME2.0', 'bouncingBall.fmu')
```
Comment (chria) [inline tests/test_log.py:116]: Should this not be: cls.fmu_path = os.path.join(get_examples_folder(), 'files', 'FMUs', 'XML', 'ME2.0', 'bouncingBall.fmu') ?

File: tests/test_log.py (hunk 2 above)
Comment (chria) [reply tests/test_log.py:109]: Ok, see comment below.

File: tests/test_log.py (hunk 3 above)
Comment (modelonrobinandersson) [reply tests/test_log.py:116]: That would be the case if it was the folder 'files' in 'tests' but now this FMU is picked from the example folder instead and the path is slightly different there.

**Comment by chria** [comment]
Looks good!

**Pull request merged**